src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs

```diff
@@ -259,10 +262,18 @@ private async Task ProcessLogInCallback()
 
     private async Task ProcessLogOut(string returnUrl)
     {
-        if (!await SignOutManager.ValidateSignOutState())
+        // We keep around SignOutManager for backwards compatibility with existing apps, but we
+        // are obsoleting that mechanism in favor of using the history API.
+        if ((Navigation.State != null && !string.Equals(Navigation.State, NavigationManagerExtensions.LogoutNavigationState, StringComparison.Ordinal) ||
+            Navigation.State == null && !await SignOutManager.ValidateSignOutState()))
         {
-            var uri = $"{Navigation.ToAbsoluteUri(ApplicationPaths.LogOutFailedPath)}?message={Uri.EscapeDataString("The logout was not initiated from within the page.")}";
```

Comment:
Please, check if "State" parameter is provided in the appropriate tests, seems like "Message" parameter is still used.